Tests/LibGfx/TestMedianCut.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/MedianCut.h>
+#include <LibTest/TestCase.h>
+
+TEST_CASE(single_element)
+{
+    auto const bitmap = TRY_OR_FAIL(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, { 1, 1 }));
+    bitmap->set_pixel(0, 0, Gfx::Color::NamedColor::White);
+
+    auto const result = TRY_OR_FAIL(Gfx::median_cut(bitmap, 1));
+
+    EXPECT_EQ(result.palette().size(), 1ul);
+    EXPECT_EQ(result.closest_color(Gfx::Color::NamedColor::White), Gfx::Color::NamedColor::White);
+}
+
+TEST_CASE(four_colors_four_indexes)
+{
+    auto const bitmap = TRY_OR_FAIL(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, { 2, 2 }));
+    bitmap->set_pixel(0, 0, Gfx::Color { 253, 0, 0 });
+    bitmap->set_pixel(1, 0, Gfx::Color { 255, 0, 0 });
+    bitmap->set_pixel(0, 1, Gfx::Color { 0, 253, 0 });
+    bitmap->set_pixel(1, 1, Gfx::Color { 0, 255, 0 });
+
+    auto const result = TRY_OR_FAIL(Gfx::median_cut(bitmap, 4));
+
+    EXPECT_EQ(result.palette().size(), 4ul);
+    EXPECT_EQ(result.closest_color(Gfx::Color(253, 0, 0)), Gfx::Color(253, 0, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(255, 0, 0)), Gfx::Color(255, 0, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(0, 253, 0)), Gfx::Color(0, 253, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(0, 255, 0)), Gfx::Color(0, 255, 0));
+}
+
+TEST_CASE(four_color_two_index)
+{
+    auto const bitmap = TRY_OR_FAIL(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, { 2, 2 }));
+    bitmap->set_pixel(0, 0, Gfx::Color { 253, 0, 0 });
+    bitmap->set_pixel(1, 0, Gfx::Color { 255, 0, 0 });
+    bitmap->set_pixel(0, 1, Gfx::Color { 0, 253, 0 });
+    bitmap->set_pixel(1, 1, Gfx::Color { 0, 255, 0 });
+
+    auto const result = TRY_OR_FAIL(Gfx::median_cut(bitmap, 2));
+
+    EXPECT_EQ(result.palette().size(), 2ul);
+    EXPECT_EQ(result.closest_color(Gfx::Color(253, 0, 0)), Gfx::Color(254, 0, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(255, 0, 0)), Gfx::Color(254, 0, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(0, 253, 0)), Gfx::Color(0, 254, 0));
+    EXPECT_EQ(result.closest_color(Gfx::Color(0, 255, 0)), Gfx::Color(0, 254, 0));
+}
```

Comment:
Maybe worth having a test where the target palette size is smaller than the number of input colors?

Reply:
It's already the case of the last test case, four colors in and two out, or am I misunderstanding you?

Reply:
Oh, sorry, yes I misread the test case!